integration-libs/opf/b2b-checkout/components/opf-b2b-checkout-review/opf-b2b-checkout-review.component.ts

```diff
@@ -0,0 +1,145 @@
+  standalone: false,
+})
+export class OpfB2bCheckoutReviewComponent
+  extends CheckoutReviewSubmitComponent
+  implements OnInit
+{
+  protected fb = inject(UntypedFormBuilder);
+  protected opfMetadataStoreService = inject(OpfMetadataStoreService);
+  protected cmsService = inject(CmsService);
+  protected checkoutPaymentTypeFacade = inject(CheckoutPaymentTypeFacade);
+  protected opfBaseFacade = inject(OpfBaseFacade);
+  protected checkoutCostCenterFacade = inject(CheckoutCostCenterFacade);
+
+  protected defaultTermsAndConditionsFieldValue = false;
+
+  checkoutStepTypePaymentType = CheckoutStepType.PAYMENT_TYPE;
+
+  explicitTermsAndConditions$: Observable<boolean | undefined> = this.cmsService
+    .getCurrentPage()
+    .pipe(
+      map((page: Page) => {
+        return this.isCmsComponentInPage(
+          OPF_EXPLICIT_TERMS_AND_CONDITIONS_COMPONENT,
+          page
+        );
+      })
+    );
+
+  checkoutSubmitForm: UntypedFormGroup = this.fb.group({
+    termsAndConditions: [
+      this.defaultTermsAndConditionsFieldValue,
+      Validators.requiredTrue,
+    ],
+  });
+
+  get termsAndConditionInvalid(): boolean {
+    return this.checkoutSubmitForm.invalid;
+  }
+
+  get termsAndConditionsFieldValue(): boolean {
+    return Boolean(this.checkoutSubmitForm.get('termsAndConditions')?.value);
+  }
+
+  get paymentType$(): Observable<PaymentType | undefined> {
+    return this.activeCartFacade
+      .getActive()
+      .pipe(map((cart: Cart) => cart.paymentType));
+  }
+
+  get poNumber$(): Observable<string | undefined> {
+    return this.checkoutPaymentTypeFacade.getPurchaseOrderNumberState().pipe(
+      filter((state) => !state.loading && !state.error),
+      map((state) => state.data)
+    );
+  }
+
+  get costCenter$(): Observable<CostCenter | undefined> {
+    return this.checkoutCostCenterFacade.getCostCenterState().pipe(
+      filter((state) => !state.loading && !state.error),
+      map((state) => state.data)
+    );
+  }
+
+  getCostCenterCard(costCenter?: CostCenter | null): Observable<Card> {
+    return combineLatest([
+      this.translationService.translate('opfCheckout.costCenter'),
+      this.translationService.translate('opfCheckout.noCostCenter'),
+    ]).pipe(
+      map(([title, noCostCenter]) => ({
+        title,
+        textBold: costCenter?.name ?? noCostCenter,
+        text: [],
+      }))
+    );
+  }
+
+  getPoNumberCard(poNumber?: string | null): Observable<Card> {
+    return combineLatest([
+      this.translationService.translate('opfCheckout.poNumber'),
+      this.translationService.translate('opfCheckout.noPoNumber'),
+    ]).pipe(
+      map(([textTitle, noneTextTitle]) => {
+        return {
+          title: textTitle,
+          textBold: poNumber ? poNumber : noneTextTitle,
+        };
+      })
+    );
+  }
+
+  protected isCmsComponentInPage(cmsComponentUid: string, page: Page): boolean {
+    return !!page && JSON.stringify(page).includes(cmsComponentUid);
```

Comment:
Would you explain why we need to use JSON.stringify here?

Reply:
The use of `JSON.stringify(page)` is a bit of a hacky approach to search for a CMS component ID within the entire page object. 

`JSON.stringify` searches through the entire page object structure, including nested properties
It can find the component ID even if it's deeply nested in the page object structure.

This approach was used mainly becauseI t's a one-liner and making the code concise. It avoids the need to understand or traverse the complex structure of the page object manually and eliminates the need to handle multiple levels of nesting.

Reply:
Sorry to jump in,
Did we try the alternative approach - picking the value from proper nested properties? If yes, what specific challenges did we face?

btw. the current solution is prone to edge-case name clashes i.e. if the same string ID used in a different property (e.g. as an ID of some cms slot - in that case the _component with such ID_ is not present of the page, but we would think it is). We don't have control over how customers name their other CMS components and slots.

Reply:
As we agreed on the slack conversation I will create a ticket to find the better solution for this.